src/api.rs

```diff
@@ -181,20 +209,22 @@ pub unsafe extern "C" fn destroy_verify_piece_inclusion_proof_response(
 /// Returns the merkle root for a piece after piece padding and alignment.
 #[no_mangle]
 pub unsafe extern "C" fn generate_piece_commitment(
-    piece_path: *const libc::c_char,
+    piece_fd: *mut libc::c_void,
```

Comment:
BLOCKING: 

Couple of things:

- This should be `libc::c_int`, not `*mut libc::c_void`.
- Who is responsible for closing the file descriptor? If it's the caller, you should say so in the comments.